sled-agent/src/rack_setup/service.rs

```diff
@@ -340,7 +343,16 @@ impl ServiceInner {
                             .collect(),
                     },
                 )
-                .await
+                .await?;
+
+                // Ensure all datasets are initialized
```

Comment:
This is the equivalent of the "blueprint executor", as controlled by RSS. "Request datasets before requesting zones".